scripts/skills/effects/rf_life_leech_effect.nut

```diff
@@ -0,0 +1,100 @@
+this.rf_life_leech_effect <- ::inherit("scripts/skills/skill", {
+	m = {
+		LifeLeechValue = 1.0,   // 1.0 equals to 100% damage leeched as life
+        LeechDistance = 1,
+		ForceHeal = false   // Temporary helper variable
+	},
+	function create()
+	{
+		this.m.ID = "effects.rf_life_leech";
+		this.m.Name = "Life Leech";
+		this.m.Description = "This character recovers Hitpoints when dealing damage to enemies with blood.";
+		this.m.Icon = "skills/status_effect_09.png";
+		this.m.IconMini = "";   // We currently have no fitting mini-icon for this
+		this.m.Overlay = "status_effect_09";
+		this.m.SoundOnUse = [
+			"sounds/enemies/vampire_life_drain_01.wav",
+			"sounds/enemies/vampire_life_drain_02.wav",
+			"sounds/enemies/vampire_life_drain_03.wav"
+		];
+		this.m.Type = ::Const.SkillType.StatusEffect;
+		this.m.Order = this.Const.SkillOrder.Perk;
+		this.m.IsActive = false;
+		this.m.IsRemovedAfterBattle = false;
+		this.m.IsStacking = false;
+	}
+
+	function getTooltip()
+	{
+		local ret = this.skill.getTooltip();
+
+        local withinText = "adjacent enemies";
+        if (this.m.LeechDistance > 1) withinText = "enemies within " + ::MSU.Text.colorGreen(this.m.LeechDistance) + " tiles";
+        ret.push({
+            id = 11,
+            type = "text",
+            icon = "ui/icons/health.png",
+            text = "Heal " + ::MSU.Text.colorGreen((this.m.LifeLeechValue * 100) + "%") + " of Hitpoint damage inflicted on " + withinText + " that have blood"
+        });
+
+		return ret;
+	}
+
+	function onCombatStarted()
+	{
+		this.m.ForceHeal = false;
+	}
+
+	function onCombatFinished()
+	{
+		this.m.ForceHeal = false;
+	}
+
+	function onTargetHit( _skill, _targetEntity, _bodyPart, _damageInflictedHitpoints, _damageInflictedArmor )
+	{
+		if (_damageInflictedHitpoints <= 0) return;
+		if (this.m.ForceHeal)      // If our attack killed the target we instead handle the leeching in here
+		{
+			this.lifesteal(_damageInflictedHitpoints);
+			this.m.ForceHeal = false;
+			return;
+		}
+		local actor = this.m.Container.getActor();
+		if (actor.getHitpoints() == actor.getHitpointsMax()) return;
+		if (_targetEntity == null || !_targetEntity.isAlive()) return;      // Otherwise we crash the game when leeching from a killing blow
+		if (_targetEntity.getCurrentProperties().IsImmuneToBleeding) return;
+		if (actor.getTile().getDistanceTo(_targetEntity.getTile()) > this.m.LeechDistance) return;
```

Comment:
Please combine all these `if` checks into one `if` check using `||` operator.

Reply:
I find those single line guard clauses much better coding practice in most cases than using long if statements or god forbid multiple nested if's that cause all the code to indent to the right outside of the screen.

Vanilla does it the same way btw. Except that they use 5 lines for each if:
![image](https://user-images.githubusercontent.com/2252464/221441461-ca021a0b-d7f0-42e6-8076-2aa445dd80d8.png)

Reply:
Long `if` clauses can be broken into multiple lines and they can be very easy to read. For example:
```squirrel
if (_script == "items/weapons/greenskins/goblin_bow" ||
    _script == "items/weapons/greenskins/goblin_crossbow" ||
    _script == "items/weapons/greenskins/goblin_heavy_bow" ||
    _script == "items/weapons/named/named_goblin_heavy_bow"
)
{
	::mods_hookNewObject(_script, function(o) {
		o.m.FatigueOnSkillUse -= 2;
	});
}
```

Reply:
As this is primarily a code style discussion, we should see what @Enduriel says.

Reply:
I strongly prefer this format
```squirrel
if (_script == "items/weapons/greenskins/goblin_bow" ||
    _script == "items/weapons/greenskins/goblin_crossbow" ||
    _script == "items/weapons/greenskins/goblin_heavy_bow" ||
    _script == "items/weapons/named/named_goblin_heavy_bow"
)
```
It achieves both purposes of having a new line for each statement while also minimizing wasted spaces. One thing to keep in mind is to make sure you continue to use only tabs for this alignment.

Reply:
I applied this way of formatting now.

But the reasons you give right here are not convincing. My approach had a new line for each statement & minimized wasted space. Even more so that this suggested approach here.
You lose a whole line for the `)` and two more lines for `{` and `}`.

The argumentation should instead have been readability or staying with a style that most are probably already used to